### PR TITLE
Frontend - update all of our npm packages

### PR DESCRIPTION
**Pull request by pavish**
This PR updates almost all of our npm packages. I've verified and fixed all package update related breaking changes. 

**Before reviewing**
* Stop the server
* Pull in branch
* Delete your local node_modules folder and the node_modules shared docker volume
* Start the server
* If you run into any problem, purge your docker container and build again

**Checklist of things to verify for reviewers**
* Run dev server. Check if page loads and test general sanity on any of the pages.
* Build with `npm run build` and run Mathesar in production mode. Check if page loads and test general sanity.
* These should be covered already by our CI:
   * `npm run typecheck`
   * `npm run test`
   * `npm run check-format`
   * `npm run lint`
* Run `npm run storybook` and `npm run build-storybook` and test general sanity on storybook.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
I did not review the file changes or the things marked "should be covered already by our CI".

### Success
- Dev server – works fine, I tried importing a file and clicking around in a table
- Production server - works fine, I tried schema page, browsing a table, editing a table 
- Storybook - seems fine, I ran `npm run storybook` and browsed around different components of the storybook.

### Failure
I ran  `docker exec -it -w /code/mathesar_ui mathesar_service npm run build-storybook`, here was the output:

```
> mathesar_ui@0.0.0 build-storybook /code/mathesar_ui
> build-storybook

info @storybook/svelte v6.5.12
info
info => Cleaning outputDir: /code/mathesar_ui/storybook-static
info => Loading presets
info Found existing addon "@storybook/addon-docs", skipping.
info => Compiling manager..
vite v3.1.3 building for production...
                                                                                                                       illed
npm ERR! code ELIFECYCLE
npm ERR! errno 137
npm ERR! mathesar_ui@0.0.0 build-storybook: `build-storybook`
npm ERR! Exit status 137
npm ERR!
npm ERR! Failed at the mathesar_ui@0.0.0 build-storybook script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-09-23T02_10_25_795Z-debug.log
```

Here are the contents of the log file mentioned:

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build-storybook' ]
2 info using npm@6.14.17
3 info using node@v14.20.0
4 verbose run-script [ 'prebuild-storybook', 'build-storybook', 'postbuild-storybook' ]
5 info lifecycle mathesar_ui@0.0.0~prebuild-storybook: mathesar_ui@0.0.0
6 info lifecycle mathesar_ui@0.0.0~build-storybook: mathesar_ui@0.0.0
7 verbose lifecycle mathesar_ui@0.0.0~build-storybook: unsafe-perm in lifecycle true
8 verbose lifecycle mathesar_ui@0.0.0~build-storybook: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/code/mathesar_ui/node_modules/.bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
9 verbose lifecycle mathesar_ui@0.0.0~build-storybook: CWD: /code/mathesar_ui
10 silly lifecycle mathesar_ui@0.0.0~build-storybook: Args: [ '-c', 'build-storybook' ]
11 silly lifecycle mathesar_ui@0.0.0~build-storybook: Returned: code: 137  signal: null
12 info lifecycle mathesar_ui@0.0.0~build-storybook: Failed to exec build-storybook script
13 verbose stack Error: mathesar_ui@0.0.0 build-storybook: `build-storybook`
13 verbose stack Exit status 137
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:400:28)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:400:28)
13 verbose stack     at maybeClose (internal/child_process.js:1088:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
14 verbose pkgid mathesar_ui@0.0.0
15 verbose cwd /code/mathesar_ui
16 verbose Linux 5.10.47-linuxkit
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build-storybook"
18 verbose node v14.20.0
19 verbose npm  v6.14.17
20 error code ELIFECYCLE
21 error errno 137
22 error mathesar_ui@0.0.0 build-storybook: `build-storybook`
22 error Exit status 137
23 error Failed at the mathesar_ui@0.0.0 build-storybook script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 137, true ]
```

**Comment by pavish** [comment]
@kgodey Looks an issue with user permission on your environment to `/mathesar_ui/storybook-static`.

Is that folder created? Does the docker user have access to it?

It needs more debugging but I don't think we have to do that in this PR.

File: mathesar_ui/package.json [hunk 1]
```diff
@@ -7,76 +7,69 @@
...
+    "ts-jest": "^29.0.1",
+    "tslib": "^2.4.0",
+    "typescript": "^4.8.3",
+    "vite": "^3.1.3"
   },
   "dependencies": {
-    "@fortawesome/free-solid-svg-icons": "^5.15.3",
-    "@popperjs/core": "^2.9.2",
-    "dayjs": "^1.11.1",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
```
Comment (seancolsen) [inline mathesar_ui/package.json:66]: Ooo, this is helpful! More icons to choose from!

File: mathesar_ui/package.json [hunk 2]
```diff
@@ -7,76 +7,69 @@
...
-    "svelte": "^3.42.4",
-    "svelte-check": "^2.2.5",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jest": "^27.0.4",
+    "eslint-plugin-promise": "^6.0.1",
+    "eslint-plugin-svelte3": "^4.0.0",
+    "jest": "^29.0.3",
+    "jest-environment-jsdom": "^29.0.3",
+    "npm": "^8.19.2",
+    "postcss": "^8.4.16",
```
Comment (seancolsen) [inline mathesar_ui/package.json:47]: Neat! Did you have anything in mind for postcss?

Comment (pavish) [reply mathesar_ui/package.json:47]: Postcss is used internally by svelte-preprocess to make the `:global` rules work. They have removed it from being installed by default in the recent version on svelte-preprocess, so we have to install it manually.

I could see us using more of postcss features, but I do not have anything in mind for it at the moment.

**Pull request merged**